src/main/webapp/app/shared/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ * Component based on Open Source Project ag-virtual-scroll
+ * https://github.com/ericferreira1992/ag-virtual-scroll
+ *
+ */
+
+import { Component, ElementRef, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, Renderer2, SimpleChanges, ViewChild } from '@angular/core';
+import { NavigationStart, Router } from '@angular/router';
+import { VirtualScrollRenderEvent } from 'app/shared/virtual-scroll/virtual-scroll-render-event.class';
+import { ITEMS_PER_PAGE } from 'app/shared/constants/pagination.constants';
+
+@Component({
+    selector: 'virtual-scroll',
+    templateUrl: './virtual-scroll.component.html',
+    styleUrls: ['virtual-scroll.style.css'],
+})
+export class VirtualScrollComponent<T extends { id?: number }> implements OnInit, OnChanges, OnDestroy {
+    @ViewChild('itemsContainer', { static: true }) private itemsContainerElRef: ElementRef<HTMLElement>;
+
+    @Input('items') public originalItems: T[] | undefined = [];
+    @Input() collapsableHtmlClassNames: string[];
+    @Input() minItemHeight: number;
+    @Input() endOfListReachedItemThreshold: number;
+    @Input() forceReload: boolean;
+
+    @Output() forceReloadChange = new EventEmitter<boolean>();
+    @Output() onItemsRender = new EventEmitter<VirtualScrollRenderEvent<T>>();
+    @Output() onEndOfOriginalItemsReached = new EventEmitter();
+
+    public prevOriginalItems: T[] = [];
+    public domTreeItems: T[] = [];
+
+    public currentScroll = 0;
+    public contentHeight = 0;
+    public paddingTop = 0;
+
+    public startIndex = 0;
+    public endIndex = 0;
+
+    private scrollIsUp = false;
+    private lastScrollIsUp = false;
+
+    previousItemsHeight: number[] = [];
+
+    scrollListener: () => void;
+    focusInListener: () => void;
+
+    constructor(private elementRef: ElementRef<HTMLElement>, private renderer: Renderer2, private router: Router) {}
+
+    /**
+     * start listening to focusin events on initialization to prevent unintentional scrolling when the user focuses into the text area of posting markdown editor
```

Comment:
You are mentioning posts a lot here but this component seems to be generic for any kind of item? Can you remove the references to `posts` in the whole component?

Reply:
I agree, removed the word post from within this file.